yt_dlp/extractor/digitalconcerthall.py

```diff
@@ -0,0 +1,201 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+import json
+
+from yt_dlp.extractor.common import InfoExtractor
+from yt_dlp.compat import (
+    compat_HTTPError,
+)
+
+from yt_dlp.utils import (
+    ExtractorError,
+    urlencode_postdata,
+)
+
+
+class DigitalConcertHallIE(InfoExtractor):
+    IE_DESC = 'DigitalConcertHall extractor'
+    _VALID_URL = r'https?://(?:www\.)?digitalconcerthall\.com/(?P<language>[a-z]+)/concert/(?P<id>[0-9]+)'
+    _LOGIN_URL = 'https://www.digitalconcerthall.com/en/login'
+    _OAUTH_URL = 'https://api.digitalconcerthall.com/v2/oauth2/token'
+    _ACCESS_TOKEN = 'none'
+    _CLIENT_SECRET = '2ySLN+2Fwb'
+    _NETRC_MACHINE = 'digitalconcerthall'
+    # if you don't login, all you will get is trailers
+    _LOGIN_REQUIRED = True
+    _TEST = {
+        'url': 'https://www.digitalconcerthall.com/en/concert/53785',
+        'md5': 'TODO: md5 sum of the first 10241 bytes of the video file (use --test)',
+        'info_dict': {
+            'id': '53785',
+            'language': 'en',
+            'ext': 'mp4',
+            'title': 'Video title goes here',
+            'thumbnail': r're:^https?://.*/images/core/Phil.*\.jpg$',
+        }
+    }
+
+    def debug_out(self, args):
+        if not self._downloader.params.get('verbose', False):
+            return
+
+        self.to_screen('[debug] %s' % args)
+
+    def _login(self):
+        username, password = self._get_login_info()
+        if username is None:
+            if self._LOGIN_REQUIRED:
+                raise ExtractorError('No login info available, needed for using %s.' % self.IE_NAME, expected=True)
+            return
+        # first get JWT token
+        data = {
+            'affiliate': 'none',
+            'grant_type': 'device',
+            'device_vendor': 'unknown',
+            'device_model': 'unknown',
+            'app_id': 'dch.webapp',
+            'app_distributor': 'berlinphil',
+            'app_version': '1.0.0',
+            'client_secret': self._CLIENT_SECRET,
+        }
+        try:
+            token_response = self._download_json(
+                self._OAUTH_URL,
+                None, 'Obtaining token', data=urlencode_postdata(data), headers={
+                    'Content-Type': 'application/x-www-form-urlencoded',
+                })
+        except ExtractorError as e:
+            msg = 'Unable to obtain token'
+            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 401:
+                resp = self._parse_json(e.cause.read().decode(), None, fatal=False)
```

Comment:
no need to use try-except
Just set `Unable to obtain token` as errnote in download_json